e2e/runner/setup_solana.go

```diff
@@ -78,4 +84,64 @@ func (r *E2ERunner) SetSolanaContracts(deployerPrivateKey string) {
 	balance, err := r.SolanaClient.GetBalance(r.Ctx, pdaComputed, rpc.CommitmentConfirmed)
 	require.NoError(r, err)
 	r.Logger.Info("initial PDA balance: %d lamports", balance.Value)
+
+	err = r.ensureSolanaChainParams()
+	require.NoError(r, err)
+}
+
+func (r *E2ERunner) ensureSolanaChainParams() error {
```

Comment:
I think it can be factorized into a reusable method to update/set chain params for any chains. You provide in the function signature the attribute to be updated. Not a blocker for this PR

Reply:
Indeed would be great to have `r.ensureChainParams(params *observertypes.ChainParams) error`